components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -203,6 +203,12 @@ export class TeamDBImpl implements TeamDB {
             throw new ResponseError(ErrorCodes.NOT_FOUND, "A team with this ID could not be found");
         }
         const membershipRepo = await this.getMembershipRepo();
+
+        const teamSize = await membershipRepo.count({ teamId, deleted: false });
+        if (teamSize <= 1) {
```

Comment:
This isn't exactly the fix we need, right? (E.g. what if you're the last Owner, and there are 3 Members -- you shouldn't be able to change your role to Member)

Reply:
Good spot. Fixed now. Will run through it once preview wakes up